File: dotnet/progressive/Models/Student.cs
using System;
using System.Collections.Generic;

namespace progressive.Models
{
    public class Student
    {
        public int ID { get; set; }
        public int GroupID { get; set; }
        public string UniCode { get; set; }
        public string SGCode { get; set; }
        public string LastName { get; set; }
        public string OtherNames { get; set; }

        public Group Group { get; set; }

        public ICollection<Progression> Progressions { get; set; }
    }
}

File: angular/progression/src/app/models/Student.ts
import { Group } from './Group';
import { Progression } from './Progression';

export interface Student {
    id: number;
    groupId: number;
    uniCode: string;
    sgCode: string|null;
    lastName: string;
    otherNames: string;
    startDate: Date|string;
    
    group: Group;

    progressions: Progression[];
}
